src/uct/ib/mlx5/dc/dc_mlx5_ep.h

```diff
@@ -677,7 +722,8 @@ uct_dc_mlx5_iface_dci_get(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
 
     ucs_assert(!iface->super.super.config.tx_moderation);
 
-    if (uct_dc_mlx5_iface_is_dci_shared(iface)) {
+    if (uct_dc_mlx5_iface_is_dci_shared(iface) ||
+        uct_dc_mlx5_is_hw_dci(iface, ep->dci)) {
```

Comment:
seems these two check are frequently used together. separate func is worth creating